server/src/main/java/org/candlepin/resource/GuestIdResource.java

```diff
@@ -154,34 +154,35 @@ protected void populateEntity(GuestId guestId, GuestIdDTO dto) {
     }
 
     /**
-     * Populates the specified entities with data from the provided DTOs.
+     * Populates the specified entities with data from the provided guestIds.
      *
      * @param entities
      *  The entities instance to populate
      *
-     * @param dtos
-     *  The DTO containing the data with which to populate the entity
+     * @param guestIds
+     *  The list of string containing the guestIds to populate the entity
      *
      * @throws IllegalArgumentException
      *  if either entity or dto are null
      */
-    protected void populateEntities(List<GuestId> entities, List<GuestIdDTO> dtos) {
+    protected void populateEntities(List<GuestId> entities, List<String> guestIds) {
         if (entities == null) {
             throw new IllegalArgumentException("the guestId model entity is null");
         }
 
-        if (dtos == null) {
-            throw new IllegalArgumentException("the guestId dto is null");
+        if (guestIds == null) {
+            throw new IllegalArgumentException("the list of guestId is null");
         }
 
-        for (GuestIdDTO dto : dtos) {
-            if (dto == null) {
+        for (String guestId : guestIds) {
+            if (guestId == null) {
                 continue;
             }
 
-            GuestId guestId = new GuestId();
-            populateEntity(guestId, dto);
-            entities.add(guestId);
+            GuestIdDTO guestIdDTO = new GuestIdDTO().guestId(guestId);
+            GuestId guestIdEntity = new GuestId();
+            populateEntity(guestIdEntity, guestIdDTO);
+            entities.add(guestIdEntity);
```

Comment:
Now that all we have is the String guestId, there is no need to create redundant GuestIdDTO objects, or call `populateEntity`. We have eliminated the need for DTOs, so this is enough:
```
entities.add(new GuestId(guestId));
```

Reply:
Understood. Corrected now.